domain/account/usecase.go

```
@@ -5,15 +5,17 @@ import (
 )
 
 type UseCase interface {
-	CreateAccount(account entities.AccountInput) (*entities.Account, error)
+	CreateAccount(account entities.AccountInput) (entities.Account, error)
 	ListAll() []entities.Account
-	Deposit(CPF string, amount float64)
+	Deposit(CPF string, amount float64) error
 	Withdraw(CPF string, amount float64) bool
+	GetBalance(accountID string) (entities.Account, error)
 }
 
 type Repository interface {
-	FindOne(CPF string) entities.Account
-	UpdateBalance(account entities.Account)
-	InsertAccount(accountInput entities.AccountInput) (*entities.Account, error)
+	FindOne(CPF string) (entities.Account, bool)
+	UpdateBalance(account entities.Account) bool
```

since those boolean returns represent the success of the operation, why not return an error instead?